src/common/stack_mmap.c

```diff
@@ -74,6 +74,88 @@ call_abt_method(void *arg, enum AbtThreadCreateType flag,
 	return rc;
 }
 
+static int compare_stack_size(const void *arg1, const void *arg2)
+{
+	struct stack_pool_by_size *size1 = (struct stack_pool_by_size *)arg1;
+	struct stack_pool_by_size *size2 = (struct stack_pool_by_size *)arg2;
+
+	if (size1->sps_stack_size < size2->sps_stack_size)
+		return -1;
+	else if (size1->sps_stack_size > size2->sps_stack_size)
+		return 1;
+	else
+		return 0;
+}
+
+void stack_pool_by_size_destroy(struct stack_pool *sp, struct stack_pool_by_size *sps)
+{
+	void *item;
+	mmap_stack_desc_t *desc;
+
+	D_ASSERT(sp->sp_nb_sizes != 0 && (!d_list_empty(&sp->sp_stack_size_list) ||
+		 			  sp->sp_nb_sizes == 1));
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs